content/doc/book/using/best-practices.adoc

```diff
@@ -0,0 +1,177 @@
+
+Refer to the link:/doc/book/pipeline/pipeline-as-code/#organization-folders[organization folders] documentation for more details.
+
+.GitHub organization folders
+video::LbXKUKQ24T8[youtube,width=800,height=420]
+
+.GitLab group folders
+video::it6TOeQ6EHg[youtube,width=800,height=420]
+
+.Bitbucket project folders
+video::85b6fiVolfk[youtube,width=800,height=420]
+
+.Gitea organization folders
+video::NO3sZWRxgQM[youtube,width=800,height=420]
+
+=== Use multibranch Pipelines
+
+If something prevents you from using organization folders, then use multibranch Pipelines.
+Organization folders are preferred over multibranch Pipelines because they will automatically create and delete multibranch projects when repositories are created or deleted.
+
+.GitHub multibranch Pipelines
+video::aDmeeVDrp0o[youtube,width=800,height=420]
+
+.GitLab group folders
+video::y4XGFluzPHY[youtube,width=800,height=420]
+
+.Bitbucket project folders
+video::LNfthmZuRDI[youtube,width=800,height=420]
+
+Refer to the link:/doc/book/pipeline/pipeline-as-code/#multibranch-pipeline-projects[multibranch Pipelines] documentation for more details.
+
+=== Use Pipeline
+
+If something prevents you from using organization folders and multibranch Pipelines, use Pipeline jobs.
+Organization folders and multibranch Pipelines are preferred over Pipeline jobs because they will automatically create and delete multibranch projects when repositories are created or deleted.
+
+Refer to the link:/doc/book/pipeline/[Pipeline] documentation for more details.
+
+.Differences between Freestyle and Pipeline in Jenkins
+video::IOUm1lw7F58[youtube,width=800,height=420]
+
+== Manage your jobs
+
+Jenkins job definitions can be managed and improved to make user interactions smoother and more productive.
+
+=== Report build results
+
+Charts and graphs show trends and patterns that can highlight project status and progress.
+Automated test results for unit tests, integration tests, and end-to-end tests can highlight brittleness or instability.
+Coverage reports can illustrate areas where automated tests are not executing.
+Compiler warning messages can be the first indication of a problem.
+Static analysis tools can report risky code or code that may have security risks.
+Performance test results can identify delays or problem areas.
+
+The plugin:warnings-ng[Warnings Next Generation] plugin provides convenient access to many reports, including:
+
+* Compiler warnnings and errors (like gcc, clang, javac, or  golang)
+* Static analysis warnings and errors (spotbugs, checkstyle, pmd, lint, cpd, or Simian)
+* Code coverage reports
+
+.How to use the Warnings Next Generation plugin
+video::tj3xYFA6Q2o[youtube,width=800,height=420]
+
+=== Build on agents
+
+Use agents to perform builds instead of running builds on the controller.
```

Comment:
```suggestion
It is recommended to use agents for performing builds instead of running them on the controller.
```

Reply:
I intentionally left this as more command form making an explicit statement of action, rather than a suggestion of possible action.